src/components/input-field/input-field.tsx

```diff
@@ -343,7 +345,8 @@ export class InputField {
         const classList = {
             'mdc-text-field': true,
             'mdc-text-field--invalid': this.isInvalid(),
-            'mdc-text-field--disabled': this.disabled || this.readonly,
+            'mdc-text-field--disabled': this.disabled,
+            'mdc-text-field--readonly': this.readonly,
```

Comment:
As far as I can find, `mdc-text-field--readonly` doesn't exist in MDC, but is something we made up, right? If so, I don't think it should be called `mdc-text-field--readonly`. The `mdc-` prefix is used by MDC specifically to avoid conflicts with class names created by consumers (us). Apart from that consideration, using the `mdc-` prefix makes keeping the code clean more difficult. Imagine if we, somewhere down the line, remove all our own styling applied to `.mdc-text-field--readonly`. We'll probably keep applying the class name, under the assumption that it comes from MDC, and therefore MDC probably applies some styling for it.

Since we define this inside the shadow-DOM for limel-text-field, I don't think we need to specify that it's meant for use with and mdc-text-field, but if we absolutely want to indicate that, it should be done in a way that minimises the risk for confusion. Something like `lime-readonly-for-mdc-text-field` for example. If not, something like `readonly` is enough.

Reply:
> I don't think it should be called mdc-text-field--readonly.

Damn right. I forgot. We usually use `lime-` prefix for these.

Reply:
If you change this, and you're happy with my fixes, I think we can merge.

Reply:
i'll fixup